pkg/daemon/ceph/config/config.go

```diff
@@ -189,7 +200,22 @@ func CreateDefaultCephConfig(context *clusterd.Context, cluster *ClusterInfo, ru
 	i := 0
 	for _, monitor := range cluster.Monitors {
 		monMembers[i] = monitor.Name
-		monHosts[i] = monitor.Endpoint
+		monIP := cephutil.GetIPFromEndpoint(monitor.Endpoint)
+
+		// This tries to detect the current port is the mon already exist
+		// This basically handles the transtion between monitors running on 6790 to msgr2
+		currentMonPort := cephutil.GetPortFromEndpoint(monitor.Endpoint)
+
+		monPorts := [2]string{"3300", currentMonPort}
```

Comment:
I think it might be good process to make a constant for the msgr2 port and make this a reference to said constant.
```go
const msgr2port = 3300
...
monPorts := [2]string{strings.Itoa(msgr2port), currentMonPort}
```